### PR TITLE
Allow setting ip address with network for power instance

### LINKED ISSUE #3102: Ability to provide IP address for provisioning Power Systems using ibm_pi_instance resource
Hi,

ibm_pi_instance terraform resource is missing the ability to provide the IP address for the Power Systems VSI. There is client who is looking for the same. Request if this could be implemented.

### PR DESCRIPTION
**Pull request by yussufsh**
Signed-off-by: Yussuf Shaikh <yussuf.shaikh@ibm.com>

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #3102

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccIBMPIInstanceNetwork'
=== RUN   TestAccIBMPIInstanceNetwork
--- PASS: TestAccIBMPIInstanceNetwork (1096.89s)
PASS
...
```

### CODE REVIEW

File: website/docs/r/pi_instance.html.markdown [hunk 1]
```diff
@@ -62,7 +64,8 @@ Review the argument references that you can specify for your resource.
 - `pi_key_pair_name` - (Required, String) The name of the SSH key that you want to use to access your Power Systems Virtual Server instance. The SSH key must be uploaded to IBM Cloud.
 - `pi_memory` - (Required, Float) The amount of memory that you want to assign to your instance in gigabytes.
 - `pi_migratable`- (Optional, Bool) Indicates the VM is migrated or not.
-- `pi_network_ids` - (Required, String) The list of network IDs that you want to assign to the instance. 
+- `pi_network_ids` - (Required, List of String) The list of network IDs that you want to assign to the instance. This attribute is **Deprecated** use `pi_network` instead.
+- `pi_network` - (Required, List of Map) List of one or more networks to attach to the instance. The pi_network object structure is documented below.
```
Comment (geethasathyamurthy) [inline website/docs/r/pi_instance.html.markdown:68]: The pi_network object structure is documented below. 
as

must be in code tag `pi_network`

File: website/docs/r/pi_instance.html.markdown [hunk 2]
```diff
@@ -74,15 +77,18 @@ Review the argument references that you can specify for your resource.
...
+  - `ip_address` - (String) The ip address to be used of this network.
 
 ## Attribute reference
 In addition to all argument reference list, you can access the following attribute reference after your resource is created.
 
-- `addresses` - Map of strings - A list of addresses that are assigned to the instance.
+- `addresses` - Map of strings - A list of addresses that are assigned to the instance. This attribute is **Deprecated** use `pi_network` instead.
 
-  Nested scheme for `addresses`:
+  Nested schema for `addresses`:
```
Comment (geethasathyamurthy) [inline website/docs/r/pi_instance.html.markdown:91]: schema should be scheme

File: website/docs/r/pi_instance.html.markdown [hunk 3]
```diff
@@ -102,6 +108,15 @@ In addition to all argument reference list, you can access the following attribu
 - `status` - (String) The status of the instance.
 - `pin_policy`  - (String) The pinning policy of the instance.
 - `progress` - (Float) - Specifies the overall progress of the instance deployment process in percentage.
+- `pi_network` - (List of Map) - A list of networks that are assigned to the instance.
+
+  Nested schema for `pi_network`:
```
Comment (geethasathyamurthy) [inline website/docs/r/pi_instance.html.markdown:113]: schema should be scheme

**Review by geethasathyamurthy — COMMENTED** [review]
Docs part is reviewed

File: website/docs/r/pi_instance.html.markdown (hunk 1 above)
Comment (yussufsh) [reply website/docs/r/pi_instance.html.markdown]: done

**Pull request merged**